### PR TITLE
Xcode 13.1 support

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

build the project with Xcode 13

### dependency
- [x] https://github.com/wireapp/wire-ios-transport/pull/223
- [x] https://github.com/wireapp/wire-ios-cryptobox/pull/73
- [x] https://github.com/wireapp/wire-ios-mocktransport/pull/191
- [x] https://github.com/wireapp/wire-ios-data-model/pull/1234
- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/371
- [x] https://github.com/wireapp/wire-ios-share-engine/pull/266
- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/1526

### TODO

- [x] rename `IconLabelButton.subtitleLabel` to prevent name conflict with iOS 15 new property
- [x] bump frameworks
- [x] https://github.com/wireapp/wire-ios/pull/5270
- [x] undo `azure-pipelines.yml` branch
- [x] release `"wireapp/Starscream" "4.0.4_xcode13"`

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -17,15 +17,15 @@ github "wireapp/swift-protobuf" "1.15.0_xcframework"
 github "wireapp/swift-snapshot-testing" "1.9.0"
 github "wireapp/wire-ios-canvas" "13.0.2"
 github "wireapp/wire-ios-cryptobox" "29.0.0"
-github "wireapp/wire-ios-data-model" "267.0.1"
+github "wireapp/wire-ios-data-model" "chore/xcode13"
 github "wireapp/wire-ios-images" "39.0.0"
 github "wireapp/wire-ios-link-preview" "33.0.0"
 github "wireapp/wire-ios-protos" "29.0.0"
```
Comment (billypchan) [inline Cartfile.resolved:23]:
```suggestion
github "wireapp/wire-ios-protos" "chore/xcode13"
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ variables:
 
 steps:
 - script: |
-    sudo xcode-select -switch /Applications/Xcode_12.4.app
+    sudo xcode-select -switch /Applications/Xcode_13.0.app
```
Comment (billypchan) [inline azure-pipelines.yml:19]:
```suggestion
    sudo xcode-select -switch /Applications/Xcode_13.1.app
```

File: fastlane/Fastfile [hunk 3]
```diff
@@ -87,14 +87,14 @@ platform :ios do
...
+        ios_version = "15.0"
         if !xcode_version.nil? 
             xcversion(version: xcode_version)
 
-            if xcode_version == "11.4"
-                ios_version = "13.4"
-            elsif xcode_version == "11.4.1"
-                ios_version = "13.4.1"
+            if xcode_version == "13.0"
+                ios_version = "15.0"
```
Comment (billypchan) [inline fastlane/Fastfile:95]:
```suggestion
            if xcode_version == "13.0"
                ios_version = "15.0"
            elsif xcode_version == "13.1"
                 ios_version = "15.0"
```

File: azure-pipelines.yml [hunk 4]
```diff
@@ -75,6 +75,6 @@ steps:
 - script: |
     fastlane post_test
   env:  
-    DEPENDENCIES_BASE_URL: "https://raw.githubusercontent.com/wireapp/wire-ios-shared-resources/master"
+    DEPENDENCIES_BASE_URL: "https://raw.githubusercontent.com/wireapp/wire-ios-shared-resources/chore/xcode13"
```
Comment (billypchan) [inline azure-pipelines.yml:78]:
```suggestion
    DEPENDENCIES_BASE_URL: "https://raw.githubusercontent.com/wireapp/wire-ios-shared-resources/master"
```

**Review by KaterinaWire — APPROVED** [review]

**Comment by KaterinaWire** [comment]
I just approved, so many snapshots are impossible to check.

File: .github/workflows/testWithiOS14_4Simulator.yml [hunk 5]
```diff
@@ -16,4 +16,10 @@ jobs:
       run: bash setup.sh
 
     - name: Run tests
-      run: xcodebuild test -scheme Wire-iOS -project Wire-iOS.xcodeproj -destination 'platform=iOS Simulator,name=iPhone 8,OS=15.0' | xcpretty && exit ${PIPESTATUS[0]}
+      run: xcodebuild test -scheme Wire-iOS -project Wire-iOS.xcodeproj -destination 'platform=iOS Simulator,name=iPhone 8,OS=14.4' | xcpretty && exit ${PIPESTATUS[0]}
```
Comment (typfel) [inline .github/workflows/testWithiOS14_4Simulator.yml:19]: `OS=14.4` ? it was `15.0` before is that correct?

Comment (typfel) [reply .github/workflows/testWithiOS14_4Simulator.yml:19]: Nevermind saw the name of the job now

Comment (typfel) [reply .github/workflows/testWithiOS14_4Simulator.yml:19]: So it was incorrect before.

**Review by typfel — APPROVED** [review]

File: .github/workflows/testWithiOS14_4Simulator.yml (hunk 5 above)
Comment (billypchan) [reply .github/workflows/testWithiOS14_4Simulator.yml:19]: This is the workflow to test the app with iOS 14 simulator so it is correct. But I will disable this action.

**Pull request merged**